mlflow/tracking/_tracking_service/client.py

```diff
@@ -231,14 +234,22 @@ def search_traces(
             order_by=order_by,
             page_token=page_token,
         )
+
+        def fn(trace_info: trace_info) -> Optional[TraceData]:
+            try:
+                trace_data = self._download_trace_data(trace_info.request_id)
+            except requests.exceptions.HTTPError as e:
```

Comment:
@harupy Will other implementations of `AbstractStore` like FileStore always throw an HTTPError with status code 404 if the trace data isn't found? I suspect not. We seem to be relying on an implementation detail that's specific to `RestStore`. Can we find a different exception / catch `Exception` instead?

Reply:
Catching Exception sounds good.

Reply:
updated